### DISCUSSION TITLE
Node Placement across Datacenters

### DISCUSSION THREAD

**Discussion by mitja** [mysticaltech/terraform-hcloud-kube-hetzner#659] [Q&A] [answered]
I want to run kube-hetzner with Longhorn, Rancher, Wireguard, and Cilium and wonder if distributing nodes across `fsn1` and `nbg1` is ok or if I should run all nodes in one DC. 

My assumption is distributing the control plane is probably ok (eg. 2 in `fsn1` and 1 in `ngb1`), but keeping Longhorn storage nodes in one dc would be better, at least for local volumes for a PostgreSQL DB.

What is your experience? Do you distribute both the control plane and the storage nodes? Do you even distribute the control plane nodes across `fsn1`, `nbg1`, and `hel1`?

**Comment by mysticaltech** [comment, marked as answer]
@mitja Distributing is perfectly fine as the Hetzner network takes care of it with or without wireguard (it's completely optional just for an extra layer of security, but not functionality).

Now if you are going to use a DB, best to use Longhorn with node storage instead of Volumes, as those tend to be slower (please search this project for more info). We recommend https://cloudnative-pg.io/ and it's lead by one of our core contributors @mnencia and it's really well built and efficient.

I would guess that the location of the storage nodes do not matter that much, as the latency is pretty low between all data centers anyway. But have a try!

**Reply by mitja** [reply]
Thank you very much @mysticaltech. I will go with Longhorn with distributed nodes across two data centers. Thank you also for recommending https://cloudnative-pg.io/ from @mnencia. This really is a great fit for my use case.
